test/test_application.py

```diff
@@ -1,20 +1,47 @@
-from fastapi.testclient import TestClient
+from dotenv import dotenv_values
+import pytest
+from src.application import app
+import psycopg
+from src.models import User, Activity
+from tools.db_operations import insert_item
+import os
 
-from src.application import app, users_id_db
 
-client = TestClient(app)
-
-
-def test_startup():
-    response = client.get("/")
+def test_client_startup(client_test) -> None:
+    response = client_test.get("/")
     assert response.status_code == 200
     assert response.text == "Hello, I'm good!"
 
 
-def test_post_user(mock_data_user):
-    response = client.post(
+def test_db_connection():
+    env_path = "db/.env"
+    if os.path.exists(env_path):
+        env_values = dotenv_values(env_path)
+        env_variables = ["POSTGRES_USER", "POSTGRES_DB", "POSTGRES_PASSWORD"]
+        for env_variable in env_variables:
+            os.environ[env_variable] = env_values.get(env_variable)
+
+    db_connection_config = {
+        "host": "localhost",
+        "dbname": os.getenv("POSTGRES_DB"),
+        "user": os.getenv("POSTGRES_USER"),
+        "password": os.getenv("POSTGRES_PASSWORD"),
+    }
+
+    try:
+        conn = psycopg.connect(**db_connection_config, autocommit=True)
+    finally:
+        conn.close()
+
+
+def test_post_user(mock_data_user, create_test_tables, client_test) -> None:
+    conn = app.state.connection_manager.connection
+    with conn.cursor() as cur:
+        cur.execute(create_test_tables)
```

Comment:
i see that you create test tables in every test? would it be easier if you create the test tables at the start of the test session and remove them once all tests are done?
to be clear, since the tables are quite simple, this is not a big deal or anything and you can do this. I was just wondering since it is the same operation in each test, maybe it would be better if it's done only once :)

Reply:
The reason I do it everytime is to reset the tables from test to test. The instructions in the "create_test_tables" fixture start with "DROP TABLE IF EXISTS users;" and "DROP TABLE IF EXISTS activities;"

Reply:
ok! i just thought you could create tables once, but reset or remove all values from them at each test. but since we are not doing anything complicated, this is also fine